### PR TITLE
Upgrade to rand-0.5

### PR DESCRIPTION
**Pull request by djc**

### CODE REVIEW

File: quinn-proto/src/endpoint.rs [hunk 1]
```diff
@@ -213,7 +213,7 @@ impl Endpoint {
                 config,
                 io: VecDeque::new(),
                 // session_ticket_buffer,
-                initial_packet_number: distributions::Range::new(0, 2u64.pow(32) - 1024),
+                initial_packet_number: distributions::Standard,
```
Comment (Ralith) [inline quinn-proto/src/endpoint.rs]: Why the semantic change here?

Comment (djc) [reply quinn-proto/src/endpoint.rs]: I figured random initial packet numbers are going away soon, anyway, so I felt I could get away with not figuring out how to make it get it right. But if you think it's important for current usage, I can spend a little time on it.

Comment (Ralith) [reply quinn-proto/src/endpoint.rs]: Should just be a matter of dropping in [Uniform](https://docs.rs/rand/*/rand/distributions/uniform/struct.Uniform.html).

Comment (djc) [reply quinn-proto/src/endpoint.rs]: Ah, don't know how I missed that this afternoon. Should be better now!

**Pull request merged**